### DISCUSSION TITLE
Certification Framework for Beckn-enabled Networks

### DISCUSSION THREAD

**Discussion by ravi-prakash-v** [beckn/protocol-specifications#190] [Ideas]
I am thinking of a process where a **Network Facilitator** publishes a set of machine readable rules for interoperability. These rules can be adopted by locally empanelled **Certification Agencies (CAs)** to develop testing and certification suites for that network. 

Network Participants (NPs) can get certified by these agencies and obtain a Digital Certificate of compliance. These certificates can be exchanged via a TLS handshake-like mechanism. Here the assumption is that there is a chain of trusts where CAs are themselves certified by a parent CA leading all the way to a Global Root just like SSL certificates have. 

NPs can download the parent and root CA certificates from a publicly accessible registry. That way any NP that is certified by a local CA can be trusted without looking up a Registry. 

The idea is to define certification framework that details the following
1. Publishing Machine Readable Rules
2. Certification suite architecture
3. Digital Certificate structure
4. Certificate exchange mechanism
5. Chain of Trust architecture

**Comment by Vedsaga** [comment]
I agree with this approach, because this could create CAs which can have expertise domain specific when it comes to performing auditing on a Network to check if they are fairly following the protocol rules or whether they are suppressing search results to benefit species business partners...  Doing this will also bring down work load for Root directory.  However,  one question so, does NPs have to pay so fees to CAs just for keeping metaince ?
